librz/asm/p/asm_hexagon.c

```diff
@@ -13,30 +13,118 @@
 #include <rz_util.h>
 #include <rz_asm.h>
 #include <rz_lib.h>
+#include <rz_util/rz_print.h>
+#include <rz_vector.h>
 #include "hexagon.h"
 #include "hexagon_insn.h"
 #include "hexagon_arch.h"
 
-/**
- * \brief Disassembles a hexagon opcode, write info to op and returns its size.
- *
- * \param a The current RzAsm struct.
- * \param op The RzAsmOp which is be filled with the reversed opcode information.
- * \param buf The buffer with the opcode.
- * \param l The size to read from the buffer.
- * \return int Size of the reversed opcode.
- */
-static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int l) {
-	rz_return_val_if_fail(a && op && buf, -1);
-	if (l < 4) {
-		return -1;
+static RZ_OWN RzPVector /* RzAsmTokenPattern */ *get_token_patterns() {
+	static RzPVector *pvec = NULL;
+	if (pvec) {
+		return pvec;
 	}
-	ut32 addr = (ut32)a->pc;
-	HexReversedOpcode rev = { .action = HEXAGON_DISAS, .ana_op = NULL, .asm_op = op };
 
-	hexagon_reverse_opcode(a, &rev, buf, addr);
+	pvec = rz_pvector_new(rz_asm_token_pattern_free);
 
-	return op->size;
+	RzAsmTokenPattern *pat = RZ_NEW0(RzAsmTokenPattern);
+	pat->type = RZ_ASM_TOKEN_META;
+	pat->pattern = strdup(
+		"(^[\\[\\?\\/\\|\\\\\\{])|(┌)|(│)|(└)|" // Packet prefix
+		"((∎)|[<\\}])([ :])(endloop[01]{1,2})" // Endloop markers
+	);
+	rz_pvector_push(pvec, pat);
+
+	pat = RZ_NEW0(RzAsmTokenPattern);
+	pat->type = RZ_ASM_TOKEN_META;
+	pat->pattern = strdup(
+		"(#{1,2})|(\\}$)|" // Immediate prefix, Closing packet bracket
+		"\\.new|:n?t|:raw|<err>" // .new and jump hints
+	);
+	rz_pvector_push(pvec, pat);
```

Comment:
For next PR or a following one... I feel these operations are not really specific to hexagon. Maybe a sort of "builder API" would help here to avoid duplication.

Something like:
```c
RzPVector *pvec = rz_asm_token_pattern_builder();
rz_asm_token_pattern_builder_add(RZ_ASM_TOKEN_META, ".....");
```

Reply:
Sounds like a great idea to me!